audioipc/src/sys/unix/cmsg.rs

```diff
@@ -43,14 +43,21 @@ pub fn encode_handle(cmsg: &mut BytesMut, handle: RawFd) {
     }
 }
 
-// Decode one cmsghdr containing a handle, and adjust the `cmsg` buffer cursor past
-// the decoded handle.
-pub fn decode_handle(cmsg: &mut BytesMut) -> RawFd {
+// Decode one cmsghdr containing handle(s), and adjust the `cmsg` buffer cursor past
+// the decoded handle(s).
+// Note: ideally this would be a single handle, but due to buffering multiple
+// sendmsgs can coalesce into a single recvmsg.  On some (64-bit) systems, the
+// minimum alignment of the cmsghdr buffer provides capacity for 2 handles, so
+// this code must expect 1 or 2 handles per decode call.
+pub fn decode_handles(cmsg: &mut BytesMut) -> (RawFd, Option<RawFd>) {
```

Comment:
Have you considered returning `Vec<RawFd>`? It could simplify the caller site code by iterating `RawFd`, although it doesn't emphasize the length is range [1, 2]